### ISSUE TITLE
feature request: trace of a DistributedMatrix

### ISSUE THREAD

**Issue by AlexGKim** [pghysels/STRUMPACK#73]
I need a method that provides the trace of a DistribuedMatrix.

**Comment by pghysels** [comment]
https://github.com/pghysels/STRUMPACK/commit/ca2768b23bb01dbd60bb682cc7e372e965e0144a

**Comment by AlexGKim** [comment]
Hi @pghysels .  Based on the behavior of the new method and my reading of the code, can I confirm whether this is the trace of the whole matrix versus that the sub matrix stored locally?

**Comment by pghysels** [comment]
Yes, sorry, I forgot to combine the local contributions. I should have tested. Can you try again with:
https://github.com/pghysels/STRUMPACK/commit/1ebb2bf7a662442a983cc641539d7168967d74ba
